src/Og.php

```diff
@@ -267,16 +268,32 @@ public static function getMemberships(AccountInterface $user, array $states = [O
    * @param string $field_name
    *   (optional) The field name associated with the group.
    *
-   * @return \Drupal\og\Entity\OgMembership|NULL
-   *   The OgMembership entity, or NULL if the user is not a member of the
-   *   group.
+   * @return \Drupal\og\Entity\OgMembership
+   *   The OgMembership entity. If the user is not a member of the group, an
+   *   OgMembership entity will be returned in which the user has the
+   *   'non-member' role.
    */
   public static function getMembership(AccountInterface $user, EntityInterface $group, array $states = [OgMembershipInterface::STATE_ACTIVE], $field_name = NULL) {
     foreach (static::getMemberships($user, $states, $field_name) as $membership) {
       if ($membership->getGroupEntityType() === $group->getEntityTypeId() && $membership->getEntityId() === $group->id()) {
         return $membership;
       }
     }
+
+    // The user is not a member, return an OgMembership entity in which the user
+    // has the 'non-member' role.
+    $role_id = implode('-', [
+      $group->getEntityTypeId(),
+      $group->bundle(),
+      OgRoleInterface::ANONYMOUS,
+    ]);
+    $storage = \Drupal::entityTypeManager()->getStorage('og_membership');
+    return $storage->create(['type' => OgMembershipInterface::TYPE_DEFAULT])
+      ->setUser($user->id())
+      ->setEntityId($group->id())
+      ->setGroupEntityType($group->getEntityTypeId())
+      ->setState(OgMembershipInterface::STATE_PENDING)
+      ->setRoles([$role_id]);
```

Comment:
Just saw that this is ignoring `$states` - if `$states` equals `[OgMembershipInterface::STATE_ACTIVE]` then this should still return `NULL` instead of a membership in which the state equals `OgMembershipInterface::STATE_PENDING`.